### PR TITLE
Feature/vault in royalty accumulator

### PR DESCRIPTION
**Pull request by iamyulong**
Adds a vault in package and component royalty accumulator substate.

### CODE REVIEW

File: radix-engine/src/model/royalty/module.rs [hunk 1]
```diff
@@ -34,6 +34,24 @@ impl Default for RoyaltyModule {
     }
 }
 
+macro_rules! pre_load_vault {
```
Comment (talekhinezh) [inline radix-engine/src/model/royalty/module.rs:37]: Can we add TODO to remove this at some point, definitely a code smell I think that we require this.

File: radix-engine/tests/transaction_commit_rollback.rs [hunk 2]
```diff
@@ -30,10 +30,10 @@ fn test_state_track_success() {
     // Assert
     receipt.expect_commit_success();
     assert_eq!(
-        22,
+        25,
```
Comment (talekhinezh) [inline radix-engine/tests/transaction_commit_rollback.rs:33]: This feels like a fragile (non-resistant to refactoring) test. Can we just remove this assertion or replace with something more resistant to refactoring?

**Review by talekhinezh — APPROVED** [review]

File: radix-engine/src/model/royalty/module.rs (hunk 1 above)
Comment (iamyulong) [reply radix-engine/src/model/royalty/module.rs:37]: Yes!

File: radix-engine/tests/transaction_commit_rollback.rs (hunk 2 above)
Comment (iamyulong) [reply radix-engine/tests/transaction_commit_rollback.rs:33]: Definitely.

**Pull request merged**